src/sys/windows/selector.rs

```diff
@@ -464,16 +481,28 @@ impl SelectorInner {
 
     unsafe fn update_sockets_events(&self) -> io::Result<()> {
         let mut update_queue = self.update_queue.lock().unwrap();
-        loop {
-            let sock = match update_queue.pop_front() {
-                Some(sock) => sock,
-                None => break,
-            };
+        let mut update_err: bool = false;
+
+        for sock in update_queue.iter_mut() {
```

Comment:
Can just use an `unwrap_or_else` here since we don't use the `Ok` value:

```
for sock in update_queue.iter_mut() {
    let mut sock_internal = sock.lock().unwrap();
    if !sock_internal.is_pending_deletion() {
        sock_internal
            .update(&sock)
            .unwrap_or_else(|_| update_err = true);
    }
}
```

Reply:
ok, thanks